### PR TITLE
Hosted Documentation

### PR DESCRIPTION
**Pull request by Saghen**
Moves the docs over to a dedicated folder with support for [doc-builder](https://github.com/huggingface/doc-builder). Adds plenty of new documentation around architecture, tools, and web search. Also attempted to standardize the documentation around the model providers.

Adds CI for previewing and publishing to https://huggingface.co/docs/chat-ui

### CODE REVIEW

**Comment by nsarrazin** [comment]
🔥 Huge fan of this, thanks a lot!

File: docs/source/installation/spaces.md [hunk 1]
```diff
@@ -0,0 +1,9 @@
+# Running on Huggingface Spaces
+
+If you don't want to configure, setup, and launch your own Chat UI yourself, you can use this option as a fast deploy alternative.
+
+You can deploy your own customized Chat UI instance with any supported [LLM](https://huggingface.co/models?pipeline_tag=text-generation&sort=trending) of your choice on [Hugging Face Spaces](https://huggingface.co/spaces). To do so, use the chat-ui template [available here](https://huggingface.co/new-space?template=huggingchat/chat-ui-template).
```
Comment (mishig25) [inline docs/source/installation/spaces.md:5]:
```suggestion
You can deploy your own customized Chat UI instance with any supported [LLM](https://huggingface.co/models?pipeline_tag=text-generation) of your choice on [Hugging Face Spaces](https://huggingface.co/spaces). To do so, use the chat-ui template [available here](https://huggingface.co/new-space?template=huggingchat/chat-ui-template).
```

trending is already the default option

File: docs/source/installation/local.md [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+You may start an instance locally for non-production use cases. For production use cases, please see the other installation options.
+
+## Configuration
+
+The default config for Chat UI is stored in the `.env` file. You will need to override some values to get Chat UI to run locally. Start by creating a `.env.local` file in the root of the repository as per the [configuration section](/configuration/overview). The bare minimum config you need to get Chat UI to run locally is the following:
+
+```ini
+MONGODB_URL=<the URL to your MongoDB instance>
+HF_TOKEN=<your access token>
```
Comment (mishig25) [inline docs/source/installation/local.md:11]:
```suggestion
HF_TOKEN=<your access token> # find your token at hf.co/settings/token
```

File: docs/source/installation/local.md [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+In which case the url of your DB will be `MONGODB_URL=mongodb://localhost:27017`.
+
+Alternatively, you can use a [free MongoDB Atlas](https://www.mongodb.com/pricing) instance for this, Chat UI should fit comfortably within their free tier. After which you can set the `MONGODB_URL` variable in `.env.local` to match your instance.
+
+## Starting the server
+
+- Install dependencies with `npm ci`
+- Build the project with `npm run build`
+- Start the server with `npm run preview`
+- Access your instance at http://localhost:4173 🎉
```
Comment (mishig25) [inline docs/source/installation/local.md:33]:
```suggestion
```bash
npm ci # install dependencies
npm run build # build the project
npm run preview -- --open # start the server with & open your instance at http://localhost:4173
```
```

**Comment by Saghen** [comment]
@mishig25 I don't have access but I setup the CI to use `HUGGINGFACE_PUSH` and `HF_DOC_BUILD_PUSH` secrets, similar to the `safetensors` repo

**Comment by mishig25** [comment]
@gary149 I will setup the needed secrets & actions, and will merge them 😄

**Pull request merged**